presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -905,6 +934,79 @@ public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle
         metastore.dropTable(session, handle.getSchemaName(), handle.getTableName());
     }
 
+    @Override
+    public ConnectorAnalyzeTableHandle beginAnalysis(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        verifyJvmTimeZone();
+        HiveTableHandle handle = (HiveTableHandle) tableHandle;
+        SchemaTableName tableName = handle.getSchemaTableName();
+
+        Table table = metastore.getTable(tableName.getSchemaName(), tableName.getTableName())
+                .orElseThrow(() -> new TableNotFoundException(tableName));
+        checkTableIsWritable(table, writesToNonManagedTablesEnabled);
```

Comment:
@jessesleeping should we do this check here? The table might be external, but it seems it shouldn't prevent us from collecting stats for it.

What do you think @arhimondr @findepi ?

Reply:
My worry is that people will need to enable `hive.non-managed-table-writes-enabled` just to collect stats for non-managed table. That could pose some security and data consistency risks. What do you think?

Reply:
`ANALYZE` only changes the metadata. I think the main reason of adding the `hive.non-managed-table-writes-enabled` was to prevent external data from modification. But since the `ANALYZE` command doesn't change the data, but only the record in the Metastore i think it is totally fine to allow `ANALYZE`.